src/main/java/org/springframework/data/redis/connection/jedis/JedisClusterConnection.java

```diff
@@ -842,12 +840,12 @@ public JedisClusterTopologyProvider(JedisCluster cluster, Duration cacheTimeout)
 		@Override
 		public ClusterTopology getTopology() {
 
-			if (cached != null && shouldUseCachedValue()) {
-				return cached;
+			JedisClusterTopology topology = cached;
+			if (shouldUseCachedValue(topology)) {
+				return topology;
```

Comment:
the previous `shouldUseCachedValue()` would no longer be called in the patch targeting `3.3.4` so I was wondering if it makes senes to keep the arrangement stable for the SR and changing it for the `3.4` version.
For 3.3.4 we could have 
```java
protected boolean shouldUseCachedValue() {
	return shouldUseCachedValue(cached);
}
```
and call the old method.

Reply:
Feel free to apply these changes in the merge.